kubert/src/admin.rs

```diff
@@ -132,23 +145,34 @@ impl Bound {
     /// Binds and runs the server on a background task, returning a handle
     pub fn spawn(self) -> Server {
         let ready = self.ready.clone();
-        let server = self
-            .server
-            .serve(hyper::service::make_service_fn(move |_conn| {
-                let ready = ready.clone();
-                future::ok::<_, hyper::Error>(hyper::service::service_fn(
-                    move |req: hyper::Request<hyper::Body>| match req.uri().path() {
-                        "/live" => future::ok(handle_live(req)),
-                        "/ready" => future::ok(handle_ready(&ready, req)),
-                        _ => future::ok::<_, hyper::Error>(
-                            Response::builder()
-                                .status(hyper::StatusCode::NOT_FOUND)
-                                .body(hyper::Body::default())
-                                .unwrap(),
-                        ),
-                    },
-                ))
-            }));
+        let metrics = self
+            .prometheus
+            .install_recorder()
+            .expect("failed to install Prometheus recorder");
+        let process = Collector::default();
+        process.describe();
+
+        let server = {
+            self.server
+                .serve(hyper::service::make_service_fn(move |_conn| {
+                    let ready = ready.clone();
+                    let metrics = metrics.clone();
+                    let process = process.clone();
+                    future::ok::<_, hyper::Error>(hyper::service::service_fn(
+                        move |req: hyper::Request<hyper::Body>| match req.uri().path() {
+                            "/live" => future::ok(handle_live(req)),
+                            "/ready" => future::ok(handle_ready(&ready, req)),
+                            "/metrics" => future::ok(handle_metrics(&metrics, &process, req)),
```

Comment:
Before merging this, we're definitely going to want to make this functionality feature-gated so that the dependencies are optional.

To do that, I think we're going to want to create a cfg-flagged `metrics` module. We'll want most of the install_recorder/initalization logic in that module.